### ISSUE TITLE
examples: require-passing-this sample marks a non-`this` scope as ❌ that the example config does not report

### ISSUE THREAD

**Issue by ren-yamanashi** [ren-yamanashi/eslint-cdk-plugin#610]
### Describe the issue

`examples/lib/require-passing-this.ts` marks a `new` expression as ❌ that the example configuration does not report.

Lines 17-18:

```ts
    // ❌ Should pass `this`
    new Bucket(bucket, "SampleBucket");
```

`examples/eslint.config.ts` (and `examples/.oxlintrc.json`) enable `awscdk/require-passing-this` with the default options, i.e. `allowNonThisAndDisallowScope: true`, so a non-`this` Construct instance as scope is allowed and only the `scope` variable is reported. Linting the file confirms it: line 36 (`new Bucket(scope, "SampleBucket")`) is the only report.

```
$ npx eslint lib/require-passing-this.ts
  36:16  error  Require passing `this` in a constructor  awscdk/require-passing-this
```

The same file marks the identical form as ✅ a few lines later (lines 32-33):

```ts
    // ✅ when allowNonThisAndDisallowScope is true, can specify non-this value
    new Bucket(bucket, "SampleBucket");
```

So the sample contradicts itself and the first block's ❌ is stale. Either the annotation on line 17 should become ✅ (or the block removed), or that block should be shown under `{ allowNonThisAndDisallowScope: false }`, where it is indeed reported. `docs/rules/require-passing-this.md` is consistent and does not need a change.

### Links

- https://github.com/ren-yamanashi/awscdk-lint/blob/main/examples/lib/require-passing-this.ts#L17-L18
- https://github.com/ren-yamanashi/awscdk-lint/blob/main/examples/lib/require-passing-this.ts#L32-L33
- https://github.com/ren-yamanashi/awscdk-lint/blob/main/examples/eslint.config.ts#L23-L26
